collector/lio.go

```diff
@@ -0,0 +1,405 @@
+		),
+
+		lioRdmcpIops: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "iops"),
+			"iSCSI Memory Copy RAMDisk backstore transport operations.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpRead: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "read"),
+			"iSCSI Memory Copy RAMDisk backstore Read in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+		lioRdmcpWrite: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, lioRdmcpSubsystem, "write"),
+			"iSCSI Memory Copy RAMDisk backstore Write in byte.",
+			[]string{"iqn", "tpgt", "lun", "rdmcp", "object"}, nil,
+		),
+	}, nil
+}
+
+// /sys/kernel/config/target/iscsi/iqn*/tpgt_*/lun/lun_*/ which link
+// back to the following
+// /sys/kernel/config/target/core/{backstoreType}_{number}/{objectName}/
+
+func (c *lioCollector) updateStat(ch chan<- prometheus.Metric, s *iscsi.Stats) error {
+
+	log.Debugf("lio updateStat iscsi %s path", s.Name)
+	tpgtS := s.Tpgt
+	for _, tpgt := range tpgtS {
+		tpgtPath := tpgt.TpgtPath
+		iscsiEnable := tpgt.IsEnable
+
+		log.Debugf("lio: iscsi %s isEnable=%t", tpgtPath, iscsiEnable)
+		// let's not putting more line into the graph with multiple
+		// disable lun, it may create problem for bigger cluster
+		if iscsiEnable {
+
+			lunS := tpgt.Luns
+			for _, lun := range lunS {
+				backstoreType := lun.Backstore
+				objectName := lun.ObjectName
+				typeNumber := lun.TypeNumber
+
+				// struct type graphLabel { iqn, tpgt, lun, store, pool,  image}
+				// label := graphLabel {iqn, tpgt, lun, backstoreType, objectName, typeNumber}
+				label := graphLabel{s.Name, tpgt.Name, lun.Name, backstoreType, objectName, typeNumber}
+
+				log.Debugf("lio: iqn=%s, tpgt=%s, lun=%s, type=%s, object=%s, typeNumber=%s",
+					s.Name, tpgt.Name, lun.Name, backstoreType, objectName, typeNumber)
+
+				switch {
+				case strings.Compare(backstoreType, "fileio") == 0:
+					if err := c.updateFileIOStat(ch, label); err != nil {
+						return fmt.Errorf("failed fileio stat : %v", err)
+					}
+					break
+				case strings.Compare(backstoreType, "iblock") == 0:
+					if err := c.updateIBlockStat(ch, label); err != nil {
+						return fmt.Errorf("failed iblock stat : %v", err)
+					}
+					break
+				case strings.Compare(backstoreType, "rbd") == 0:
+					if err := c.updateRBDStat(ch, label); err != nil {
+						return fmt.Errorf("failed rbd stat : %v", err)
+					}
+					break
+				case strings.Compare(backstoreType, "rdmcp") == 0:
+					if err := c.updateRDMCPStat(ch, label); err != nil {
+						return fmt.Errorf("failed rdmcp stat : %v", err)
+					}
+					break
+				default:
+					continue
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// /sys/kernel/config/target/core/fileio_{typeNumber}/{object}/
+// udev_path has the file name
+func (c *lioCollector) updateFileIOStat(ch chan<- prometheus.Metric, label graphLabel) error {
+
```

Comment:
Similar, these extra newlines after the start of the function.

Reply:
Removed, pushed.